io_stream.c

```diff
@@ -141,6 +187,15 @@ void* php_mongo_io_stream_connect(mongo_con_manager *manager, mongo_server_def *
 				return NULL;
 			}
 		} else {
+#if PHP_VERSION_ID < 50600
+			if (verify && zend_is_true(*verify)) {
+				if (php_mongo_verify_hostname(manager, stream, server TSRMLS_CC) == FAILURE) {
+					*error_message = strdup("Hostname doesn't match");
```

Comment:
This message should include the expected hostname, and the given hostname - and possibly also that this is due to the SSL certificate.